[Event Log] Extended README.md with the documentation for a REST API and Start plugin contract. (#92562)
elastic/kibana: x-pack/plugins/event_log/README.md

It's anticipated that more "helper" methods like this will be provided in the
future.

### Start
```typescript

export interface IEventLogClientService {
  getClient(request: KibanaRequest): IEventLogClient;
}

export interface IEventLogClient {
  findEventsBySavedObjectIds(
    type: string,
    ids: string[],
    options?: Partial<FindOptionsType>
  ): Promise<QueryEventsBySavedObjectResult>;
}
```

> The plugin exposes an `IEventLogClientService` object to plugins that pre-req it.

Reviewer: **nit:** I'm not sure if my recommendation would be clearer?

```suggestion
The plugin exposes an `IEventLogClientService` object to plugins that request it.
```

Reviewer 2: ++

> Those plugins need to call `getClient(request)` to get the event log client.

Reviewer 2: ```suggestion
These plugins must call `getClient(request)` to get the event log client.
```

## Experimental RESTful API

> Using of the event log allows you to retrive the events for a given saved object type by the specified set of IDs.

Reviewer: ```suggestion
Usage of the event log allows you to retrieve the events for a given saved object type by the specified set of IDs.
```

> API listed below is experimental and could be changed or removed in the future.

Reviewer 2: ```suggestion
The following API is experimental and can change or be removed in a future release.
```

### `GET /api/event_log/{type}/{id}/_find`: Get events for a given saved object type by the ID

> Collects events information from the event log for the selected saved object by type and ID.

Reviewer: ```suggestion
Collects event information from the event log for the selected saved object by type and ID.
```

Params:

|Property|Description|Type|
|---|---|---|
|type|The type of the saved object whose events you're trying to get.|string|
|id|The id of the saved object.|string|

Query:

|Property|Description|Type|
|---|---|---|
|page|The page number.|number|

> |per_page|The number of alerts to return per page.|number|

Reviewer: ```suggestion
|per_page|The number of events to return per page.|number|
```

|sort_field|Sorts the response. Could be an event fields returned in the response.|string|
|sort_order|Sort direction, either `asc` or `desc`.|string|

> |filter|A <<kuery-query, KQL>> string that you filter with an attribute from the event. It should look like event.action: "execute".|string|

Reviewer: ```suggestion
|filter|A KQL string that you filter with an attribute from the event. It should look like event.action: "execute".|string|
```

Reviewer 2: Use code font for `event.action: "execute"`

> |start|The date to start looking for saved object events in the event log. Either an ISO date string, or a duration string indicating time since now.|string|

Reviewer 2: ```suggestion
|start|The date to start looking for saved object events in the event log. Either an ISO date string, or a duration string that indicates the time since now.|string|
```

> |end|The date to end looking for saved object events in the event log. Either an ISO date string, or a duration string indicating time since now.|string|

Reviewer 2: ```suggestion
|end|The date to stop looking for saved object events in the event log. Either an ISO date string, or a duration string that indicates the time since now.|string|
```

### `POST /api/event_log/{type}/_find`: Retrive events for a given saved object type by the IDs

> Collects events information from the event log for the selected saved object by type and a set of the IDs.

Reviewer: ```suggestion
Collects event information from the event log for the selected saved object by type and a set of the IDs.
```

Reviewer 2: "set of IDs" or simply "by IDs"

Params:

|Property|Description|Type|
|---|---|---|
|type|The type of the saved object whose events you're trying to get.|string|

Query:

|Property|Description|Type|
|---|---|---|
|page|The page number.|number|

> |per_page|The number of alerts to return per page.|number|

Reviewer: ```suggestion
|per_page|The number of events to return per page.|number|
```

> |sort_field|Sorts the response. Could be an event fields returned in the response.|string|

Reviewer 2: ```suggestion
|sort_field|Sorts the response. Could be an event field returned in the response.|string|
```

|sort_order|Sort direction, either `asc` or `desc`.|string|

> |filter|A <<kuery-query, KQL>> string that you filter with an attribute from the event. It should look like event.action: "execute".|string|

Reviewer: ```suggestion
|filter|A KQL string that you filter with an attribute from the event. It should look like event.action: "execute".|string|
```

Reviewer 2: Use code font for `event.action: "execute"`

> |start|The date to start looking for saved object events in the event log. Either an ISO date string, or a duration string indicating time since now.|string|

Reviewer 2: ```suggestion
|start|The date to start looking for saved object events in the event log. Either an ISO date string, or a duration string that indicates the time since now.|string|
```

> |end|The date to end looking for saved object events in the event log. Either an ISO date string, or a duration string indicating time since now.|string|

Reviewer 2: ```suggestion
|end|The date to stop looking for saved object events in the event log. Either an ISO date string, or a duration string that indicates the time since now.|string|
```
